### PR TITLE
feat: locks clicks and keydowns during update

### PR DESCRIPTION
**Pull request by dwmorrin**
fixes issue #79

### CODE REVIEW

**Comment by michaelphagen** [comment]
If a conflict occurs and the user clicks "close" to keep the form open, the buttons remain locked and the user is forced to refresh the browser window

File: js/app/pages/Form.html [hunk 1]
```diff
@@ -275,6 +275,13 @@
     }
     app.modal.handleStudentNote(student);
   },
+  isLocked: function() {
+    return app.pages.form.locked;
+  },
+  lock: function() {
+    app.pages.form.locked = true;
+  },
+  locked: false,
```
Comment (michaelphagen) [inline js/app/pages/Form.html]: What is this doing here? If it's an element, shouldn't it be up in the elements section?

**Comment by dwmorrin** [comment]
Last commit handles the one case you pointed out.

Are there other cases?

**Comment by dwmorrin** [comment]
Inching closer on this one.  Now it checks to make sure the single view is being shown and added .unlock() to the hide function.  Making sure we don't get "locked out".

**Pull request merged**